server.go

```diff
@@ -557,6 +558,7 @@ type serverPeer struct {
 func newServerPeer(s *server, isPersistent bool) *serverPeer {
 	return &serverPeer{
 		server:         s,
+		banManager:     s.banManager,
```

Comment:
I would not put this on every peer.  It takes up additional space for each one and isn't really needed.  Instead, I'd suggest just updating `addBanScore` to be something like:

```Go
func (sp *serverPeer) addBanScore(persistent, transient uint32, reason string) bool {
	return sp.s.banManager.AddBanScore(sp.Peer, persistent, transient, reason)
}
```

That way the rest of the code can even stay the same.